src/unearth/auth.py

```diff
@@ -284,27 +284,26 @@ def _should_save_password_to_keyring(self) -> bool:
             return False
         return input("Save credentials to keyring [y/N]: ") == "y"
 
-    def handle_400s(self, resp: Response, **kwargs: Any) -> Response:
-        # We only care about 401, 403, 404 responses, anything else we want to just
+    def handle_401(self, resp: Response, **kwargs: Any) -> Response:
+        # We only care about 401 response, anything else we want to just
         #   pass through the actual response
-        if resp.status_code not in [401, 403, 404]:
+        if resp.status_code != 401:
             return resp
 
         parsed = urlparse(cast(str, resp.url))
 
         # Query the keyring for credentials:
         username, password = self._get_new_credentials(
             resp.url,
-            allow_netrc=True,
+            allow_netrc=False,
             allow_keyring=True,
```

Comment:
We can allow keyring only for 401 errors:
```suggestion
            allow_keyring=resp.status_code==401,
```